evennia/contrib/events/scripts.py

```diff
@@ -559,8 +558,12 @@ class methods, objects inside an instance and so on will
         """
         now = datetime.now()
         delta = timedelta(seconds=seconds)
-        task_id = self.db.task_id
-        self.db.task_id += 1
+
+        # Choose a free task_id
+        used_ids = list(self.db.tasks.keys())
```

Comment:
Converting this to a list seems unnecessary and even inefficient. For a large list, querying content is an `O(N)` operation while querying a dict remains constant, that is an `O(1)` behavior. So removing the `list()` conversion here sounds like the way to go.